### PR TITLE
Fix crash in MEF cache deserialization

### PR DESCRIPTION
**Pull request by mhutch**
Fixes a crash that happened on my machine because for some reason the MEF cache control deserialized as null. Guessing it was because of a format change?

### CODE REVIEW

**Comment by Therzok** [comment]
Hmmm, we should be able to unit test this. What were the control file contents?

File: main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Composition/CompositionManager.Caching.cs [hunk 1]
```diff
@@ -117,6 +115,13 @@ internal bool CanUse ()
 						return false;
 					}
 
+					//this can return null (if the cache format changed?). clean up and start over.
```
Comment (Therzok) [inline main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Composition/CompositionManager.Caching.cs]: I guess the saveTask was cancelled and it only partly serialized. Not sure how to control that better though.

**Comment by mhutch** [comment]
Unfortunately I forgot to save it :( I did look at the file though, and it was not empty.

**Comment by mhutch** [comment]
What would you want to test in this case?

File: main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Composition/CompositionManager.Caching.cs (hunk 1 above)
Comment (Therzok) [reply main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Composition/CompositionManager.Caching.cs]: So, what I suggest here is:

* Set all the members and classes we have as required via `[Required]`. That will cause json to throw in case we don't have values.
* Add some tests with mef-cache-control containing just `{}`, and empty file, and partial json

Here is a sample JSON to create partial data:
https://gist.github.com/Therzok/fd0fae2c79bf586a8bc43f219c350e0c

**Comment by Therzok** [comment]
So, what I suggest here is:

* Set all the members and classes we have as required via `[Required]`. That will cause json to throw in case we don't have values.
* Add some tests with mef-cache-control containing just `{}`, and empty file, and partial json

Here is a sample JSON to create partial data:
https://gist.github.com/Therzok/fd0fae2c79bf586a8bc43f219c350e0c
